client/JS/giphyDom.js

```diff
@@ -0,0 +1,23 @@
+const container = document.querySelector(".main");
+const createCards = (data) => {
+  container.textContent = "";
+  data.data.forEach((element) => {
+    let card = document.createElement("div");
+    card.classList.add("giphy");
+
+    let img = document.createElement("img");
+    img.classList.add("img-giphy");
+    img.setAttribute("src", `${element.images.downsized.url}`);
+    card.appendChild(img);
+
+    container.appendChild(card);
+  });
+};
+console.log(window.location);
+const searchParams = new URLSearchParams(window.location.search);
+const anime = searchParams.get("anime");
+
+fetch(`/giphy/${anime}`)
+  .then((response) => response.json())
+  .then((response) => createCards(response))
+  .catch((err) => console.log(err));
```

Comment:
what if the fetch fails then `createCards` will not receive the expected value for `data` and will cause some error, so you should handle such cases by displaying a nice message for the user ex `no result found` `try another search term`

**NOTE** this thing about how the fetch api deals with request/response errors
```js
fetch(url).then((response) => {
  // Always gets a response, unless there is network error
  // It never throws an error for 4xx or 5xx response 😟
 // so here we need to check for the response status and display the result if it is 2xx or display error message if it is 4xx or 5xx
}).catch((error) => {
  // Only network error comes here
});
```